src/symbols_linux.cpp

```diff
@@ -64,6 +65,33 @@ static void applyPatch(CodeCache* cc) {}
 
 #endif
 
+static const void* getMainPhdr() {
+    void* main_phdr = NULL;
+    dl_iterate_phdr([](struct dl_phdr_info* info, size_t size, void* data) {
+        *(const void**)data = info->dlpi_phdr;
+        return 1;
+    }, &main_phdr);
+    return main_phdr;
+}
+
+static const char* getLdBase() {
+    const char* ld_base = (const char*)getauxval(AT_BASE);
+    if (!ld_base) {
+        Log::warn("Cannot determine base address of the loader");
+    }
+    return ld_base;
+}
+
+static const void* _main_phdr = getMainPhdr();
+static const char* _ld_base = getLdBase();
```

Comment:
I'd just inline `getauxval` call here.
Log might not be initialized at this point. I can't see `getauxval` failing anyway.

Reply:
b5d4e19224fba4fca4a9367276959207ec952dbb